File: components/collaboration/internal/metrics.cc
// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "components/collaboration/internal/metrics.h"

#include "base/metrics/histogram_functions.h"

namespace collaboration::metrics {

void RecordJoinEvent(CollaborationServiceJoinEvent event) {
  base::UmaHistogramEnumeration("CollaborationService.JoinFlow", event);
}

void RecordShareOrManageEvent(CollaborationServiceShareOrManageEvent event) {
  base::UmaHistogramEnumeration("CollaborationService.ShareOrManageFlow",
                                event);
}

}  // namespace collaboration::metrics

File: chrome/browser/resources/chromeos/accessibility/switch_access/metrics.js
// Copyright 2017 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import {StringUtil} from '../common/string_util.js';

/**
 * Class to record metrics for Switch Access.
 */
export const SwitchAccessMetrics = {
  /**
   * @param {string} menuAction
   */
  recordMenuAction: menuAction => {
    const metricName = 'Accessibility.CrosSwitchAccess.MenuAction.' +
        StringUtil.toUpperCamelCase(menuAction);
    chrome.metricsPrivate.recordUserAction(metricName);
  },
};
